tests/queries/tests.py

```diff
@@ -3690,6 +3690,58 @@ def test_ticket_22429(self):
         self.assertSequenceEqual(queryset, [st2])
 
 
+class Ticket28297Tests(TestCase):
+    def test_join_use_correct_table_28297(self):
+        """
+        Query was assigning wrong table.
+        It should get the last table created by the join
+        """
+        i1 = Ingredient.objects.create(name="tomato")
+        i2 = Ingredient.objects.create(name="lettuce")
+        i3 = Ingredient.objects.create(name="oil")
+        i4 = Ingredient.objects.create(name="flour")
+        i5 = Ingredient.objects.create(name="water")
+        i6 = Ingredient.objects.create(name="yeast")
+
+        sr1 = StepRecipe.objects.create()
+        sr1.ingredients.add(i1)
+        sr1.ingredients.add(i2)
+        sr1.ingredients.add(i3)
+
+        sr2 = StepRecipe.objects.create()
+        sr2.ingredients.add(i3)
+        sr2.ingredients.add(i4)
+        sr2.ingredients.add(i5)
+        sr2.ingredients.add(i6)
+
+        recipe1 = Recipe.objects.create(name='salad')
+        recipe1.steps.add(sr1)
+
+        recipe2 = Recipe.objects.create(name='bread')
+        recipe2.steps.add(sr2)
+
+        ingredients = ["tomato", "lettuce", "oil"]
+
+        queryset = Recipe.objects.all()
+        self.assertIs(len(queryset), 2)
+
+        queryset = queryset.annotate(total=Count('steps__ingredients', distinct=True))
+        # Make sure the number of ingredients are correct
+        self.assertIs(queryset[0].total, 3)
+        self.assertIs(queryset[1].total, 4)
+        queryset = queryset.filter(steps__ingredients__name__in=ingredients)
+        # Make sure both recipes return after query
+        self.assertIs(len(queryset), 2)
```

Comment:
Nitpick, but shouldn't these be assertEqual()? This would be consistent with `if len(queryset) == 2` instead of `if len(queryset) is 2`.